src/View/AbstractDirectoryView.vala

```diff
@@ -448,16 +448,19 @@ namespace FM {
             GLib.File? focus = focus_location != null ? focus_location.dup () : null;
 
             /* Because the Icon View disconnects the model while loading, we need to wait until
-             * the tree is thawed and the model reconnected before selecting the files */
-            Idle.add_full (GLib.Priority.LOW, () => {
+             * the tree is thawed and the model reconnected before selecting the files.
+             * Using a timeout helps ensure that the files appear in the model before selecting. Using an Idle
+             * sometimes results in the pasted file not being selected because it is not found yet in the model. */
+
+            Timeout.add (20, () => {
```

Comment:
Is there a better way to do this rather than polling for `tree_frozen` every 20ms?
Is it possible to make `tree_frozen` a property and then connect to it here: `notify["tree-frozen"].connect (() => ...`, of course disconnecting from it when we're done?

Reply:
That could work.  I'll try it.

Reply:
The problem seems to be that the tree is being unfrozen before the files are retrievable from the model.  The hacky timeout gets around that but I'll try to cure the deeper cause if possible.